gst/nnstreamer/tensor_filter/tensor_filter_common.c

```diff
@@ -2113,11 +2119,50 @@ gst_tensor_filter_common_get_property (GstTensorFilterPrivate * priv,
   return TRUE;
 }
 
+/**
+ * @brief Configure input tensor info with combi option.
+ */
+gboolean
+gst_tensor_filter_common_get_combined_in_info (GstTensorFilterPrivate * priv,
+    const GstTensorsInfo * in, GstTensorsInfo * combined)
+{
+  GList *list;
+  guint i, idx = 0;
+
+  g_return_val_if_fail (in != NULL, FALSE);
+  g_return_val_if_fail (combined != NULL, FALSE);
+
+  gst_tensors_info_init (combined);
+
+  if (priv->combi.in_combi_defined) {
+    for (list = priv->combi.in_combi; list != NULL; list = list->next) {
+      i = GPOINTER_TO_UINT (list->data);
+
+      if (i >= in->num_tensors) {
+        nns_loge ("Invalid input index %u, failed to combine info.", i);
+        goto error;
+      }
+
+      gst_tensor_info_copy (&combined->info[idx++], &in->info[i]);
```

Comment:
How about adding error handling to ensure `idx` is in the valid range.

Reply:
add condition to check max size (line 2148)